pennylane/operation.py

```diff
@@ -232,6 +232,93 @@ def expand_matrix(base_matrix, wires, wire_order=None):
     return mat
 
 
+def _local_sparse_swap_mat(i, n, format="csr"):
+    """Helper function which generates the sparse matrix of SWAP
+    for qubits: i <--> i+1 with final shape (2**n, 2**n)."""
+    assert i < n - 1
+    swap_mat = csr_matrix([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
+
+    if i == 0:
+        return kron(swap_mat, eye(2 ** (n - 2)), format=format)  # 2 + (n - 2) = n
+    if i == n - 2:
+        return kron(eye(2 ** (n - 2)), swap_mat, format=format)  # (n - 2) + 2 = n
```

Comment:
It seems that these two cases are correctly handled by the return statement below. Is there any reason to keep this?

Reply:
Nice catch, actually when I was writing the helper functions I didn't know if `eye(0)` would behave well so I add those two edge cases in preemptively.